src/components/App/App.jsx

```diff
@@ -0,0 +1,412 @@
+        }, 400);
+      });
+  }
+
+  //// API REQUEST HANDLERS ////
+
+  // User Handlers //
+  const handleRegistration = ({ email, password, name }) => {
+    const requestRegistration = () => {
+      return signup({ email, password, name }).then(() =>
+        handleLogin({ email, password })
+      );
+    };
+    return handleSubmit(requestRegistration);
+  };
+
+  const handleLogin = ({ email, password }) => {
+    const requestLogin = () => {
+      return signin({ email, password })
+        .then((res) => {
+          localStorage.setItem("jwt", res.token);
+          setUserToken(res.token);
+          return res.token;
+        })
+        .then((token) => {
+          return authorizeToken(token);
+        })
+        .then((user) => {
+          setCurrentUser(user.data);
+          setUserBookIndex(user.data.page);
+          setIsLoggedIn(true);
+        });
+    };
+    return handleSubmit(requestLogin);
+  };
+
+  const handleLogout = () => {
+    localStorage.removeItem("jwt");
+    setIsLoggedIn(false);
+    setUserToken("");
+  };
+
+  const handleEditProfile = (data) => {
+    const filteredData = Object.fromEntries(
+      Object.entries(data).filter(
+        ([key, value]) => value !== undefined && value !== currentUser[key]
+      )
+    );
+    const requestEditProfile = () => {
+      return patchUser(filteredData, userToken).then((user) => {
+        setCurrentUser(user.data);
+        if (filteredData.name) {
+          handleGetPageComments(userBookIndex + 1);
+        }
+      });
+    };
+
+    return handleSubmit(requestEditProfile);
+  };
+  // Comment Handlers //
+  const handleAddComment = ({ comment }) => {
+    const requestAddComment = () => {
+      return createComment(
+        { name: currentUser.name, comment, page: userBookIndex + 1 },
+        userToken
+      )
+        .then((post) => {
+          setPageComments([...pageComments, post]);
+          setUserComments([...userComments, post]);
+        })
+
+        .catch((error) => {
+          console.error("Error adding comment:", error);
+        });
+    };
+
+    return handleSubmit(requestAddComment);
+  };
+
+  const handleEditComment = ({ comment }) => {
+    const requestEditComment = () => {
+      return patchComment({ comment, _id: commentToEdit._id }, userToken)
+        .then((comment) => {
+          const updateComments = (commentsArray, updatedComment) => {
+            return commentsArray.map((comment) =>
+              comment._id === updatedComment._id
+                ? { ...comment, comment: updatedComment.comment }
+                : comment
+            );
+          };
+          setPageComments((prevPageComments) =>
+            updateComments(prevPageComments, comment)
+          );
+          setUserComments((prevUserComments) =>
+            updateComments(prevUserComments, comment)
+          );
+        })
+        .catch((error) => {
+          console.error("Error editing comment:", error);
+        });
```

Comment:
`catch` is not needed if you use `handleSubmit` because it contains `catch` inside it


**Please, correct it everywhere in the code**